opm/simulators/wells/WellGroupHelpers.cpp

```diff
@@ -788,120 +788,124 @@ namespace WellGroupHelpers
             return {};
         }
 
-        // Fixed pressure nodes of the network are the roots of trees.
-        // Leaf nodes must correspond to groups in the group structure.
-        // Let us first find all leaf nodes of the network. We also
-        // create a vector of all nodes, ordered so that a child is
-        // always after its parent.
-        std::stack<std::string> children;
-        std::set<std::string> leaf_nodes;
-        std::vector<std::string> root_to_child_nodes;
-        children.push(network.root().name());
-        while (!children.empty()) {
-            const auto node = children.top();
-            children.pop();
-            root_to_child_nodes.push_back(node);
-            auto branches = network.downtree_branches(node);
-            if (branches.empty()) {
-                leaf_nodes.insert(node);
-            }
-            for (const auto& branch : branches) {
-                children.push(branch.downtree_node());
+        std::map<std::string, double> node_pressures;
+        auto roots = network.roots();
+        for (const auto& root : roots) {
```

Comment:
If I did not see the code wrong, the following code is the only change, right?
```c++
        auto roots = network.roots();
        for (const auto& root : roots) {
```

Reply:
Hi, yes, that is correct. Sorry, should have commented on that, didn't realize the diff was so messy.